internal/api/v20240610preview/hcpopenshiftclusters_methods.go

```diff
@@ -315,11 +317,47 @@ func (c *HcpOpenShiftClusterResource) Normalize(out *api.HCPOpenShiftCluster) {
 // validateStaticComplex performs more complex, multi-field validations than
 // are possible with struct tag validation. The returned CloudErrorBody slice
 // contains structured but user-friendly details for all discovered errors.
-func validateStaticComplex(normalized *api.HCPOpenShiftCluster, updating bool) []arm.CloudErrorBody {
+func validateStaticComplex(normalized *api.HCPOpenShiftCluster) []arm.CloudErrorBody {
 	var errorDetails []arm.CloudErrorBody
-
-	// FIXME Perform user-assigned managed identity validation checks.
-
+	//Generate a slice of Resource IDs of ControlplaneOperators MI only , disregard the DataPlaneOperators.
+	var cpmiResourceIDs []string
+	if normalized.Identity.UserAssignedIdentities != nil {
+		for _, operatorResourceID := range normalized.Properties.Spec.Platform.OperatorsAuthentication.UserAssignedIdentities.ControlPlaneOperators {
+			cpmiResourceIDs = append(cpmiResourceIDs, operatorResourceID)
+		}
+		smiResourceID := normalized.Properties.Spec.Platform.OperatorsAuthentication.UserAssignedIdentities.ServiceManagedIdentity
+		miCount := 0
+		for resourceID, _ := range normalized.Identity.UserAssignedIdentities {
+
+			if slices.Contains(cpmiResourceIDs, resourceID) || resourceID == smiResourceID {
+				//Duplicated Identity Validation
+				if slices.Contains(cpmiResourceIDs, resourceID) && resourceID == smiResourceID {
+					errorDetails = append(errorDetails, arm.CloudErrorBody{
+						Message: fmt.Sprintf(
+							"User-assigned managed identity %s can not be part of the Operator Authentication Identities and Service Managed Identity .",
+							resourceID),
+					})
+				}
+				miCount++
+			} else {
+				errorDetails = append(errorDetails, arm.CloudErrorBody{
+					Message: fmt.Sprintf(
+						"User-assigned managed identity %s is not part of the Operator Authentication Identities.",
+						resourceID),
+				})
+			}
+		}
+		// Ensure that if User-assigned managed identity count is always equal to count of ControlPlaneOperator MIs and Service Managed Identity.count.
+		var smiCount int
+		if smiResourceID != "" {
+			smiCount = 1
+		}
+		if miCount != (len(cpmiResourceIDs) + smiCount) {
+			errorDetails = append(errorDetails, arm.CloudErrorBody{
+				Message: "User-assigned managed identity count mistmatch with the Operator Authentication Identities",
+			})
+		}
```

Comment:
I'd suggest separate error messages for `miCount < len(cpmiResourceIDs)` and `miCount > len(cpmiResourceIDs)` as well as a separate error message for `serviceManagedIdentity` not appearing in `identity.userAssignedIdentities`.

That way you can use `arm.CloudErrorBody.Target` to clarify which field has the extraneous identity(s).